vendor/k8s.io/kube-openapi/pkg/common/common.go

```diff
@@ -90,27 +90,28 @@ type Config struct {
 }
 
 var schemaTypeFormatMap = map[string][]string{
-	"uint":        {"integer", "int32"},
-	"uint8":       {"integer", "byte"},
-	"uint16":      {"integer", "int32"},
-	"uint32":      {"integer", "int64"},
-	"uint64":      {"integer", "int64"},
-	"int":         {"integer", "int32"},
-	"int8":        {"integer", "byte"},
-	"int16":       {"integer", "int32"},
-	"int32":       {"integer", "int32"},
-	"int64":       {"integer", "int64"},
-	"byte":        {"integer", "byte"},
-	"float64":     {"number", "double"},
-	"float32":     {"number", "float"},
-	"bool":        {"boolean", ""},
-	"time.Time":   {"string", "date-time"},
-	"string":      {"string", ""},
-	"integer":     {"integer", ""},
-	"number":      {"number", ""},
-	"boolean":     {"boolean", ""},
-	"[]byte":      {"string", "byte"}, // base64 encoded characters
-	"interface{}": {"object", ""},
+	"uint":                                         {"integer", "int32"},
+	"uint8":                                        {"integer", "byte"},
+	"uint16":                                       {"integer", "int32"},
+	"uint32":                                       {"integer", "int64"},
+	"uint64":                                       {"integer", "int64"},
+	"int":                                          {"integer", "int32"},
+	"int8":                                         {"integer", "byte"},
+	"int16":                                        {"integer", "int32"},
+	"int32":                                        {"integer", "int32"},
+	"int64":                                        {"integer", "int64"},
+	"byte":                                         {"integer", "byte"},
+	"float64":                                      {"number", "double"},
+	"float32":                                      {"number", "float"},
+	"bool":                                         {"boolean", ""},
+	"time.Time":                                    {"string", "date-time"},
+	"string":                                       {"string", ""},
+	"integer":                                      {"integer", ""},
+	"number":                                       {"number", ""},
+	"boolean":                                      {"boolean", ""},
+	"[]byte":                                       {"string", "byte"}, // base64 encoded characters
+	"interface{}":                                  {"object", ""},
+	"k8s.io/apimachinery/pkg/runtime.RawExtension": {"object", ""},
```

Comment:
I don't like having `RawExtension` referenced in kube-openapi, since `RawExtension` isn't part of it, but I don't see a simple way of getting around this. Any ideas?

Reply:
what if we add `common.GetOpenAPITypeFormat` to `common.Config`? We can then customize it in main repo and pass it using the config struct.

Reply:
AFAICT, `common.Config` isn't used in the generator (`generators/openapi.go`).

Reply:
Hmm. Good point. Then they need to be a customization list (in addition to the list in `common.GetOpenAPITypeFormat`) either coming from command line or in [this file](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/cmd/openapi-gen/main.go). I would prefer the later. wdyt?

Reply:
I would support loading from the file @mbohlool referenced for now; we can load from a CLI flag in the future if we need to.